java/test/jmri/jmrit/logix/WarrantTest.java

```diff
@@ -223,7 +223,8 @@ public void testWarrant() throws JmriException {
 
         // wait for done
         JUnitUtil.waitFor(() -> {
-            return Bundle.getMessage("Idle").equals(warrant.getRunningMessage());
+            String runningMsg = warrant.getRunningMessage();
+            return ( runningMsg==null?false:Bundle.getMessage("Idle").equals(runningMsg));
```

Comment:
> Reduce chances of nullpointer in waitFor
> 
> Recent failure at https://builds.jmri.org/jenkins/job/development/job/builds/1484/

I don't get how this change could stop a NPE. `String.equals(null)` shouldn't cause a NPE.

Maybe `warrant` could be null and that you need to test for that instead. But I don't think `warrant` could be null in this case. Maybe the method `getRunningMessage()` throws a NPE within that method?

Reply:
A suggestion is to make JUnitUtil.waitFor() to also log the cause exception in case of an exception.

Reply:
See #11603 for improvements to JUnitUtil.waitFor() exception cause reporting